mockgen/import_mode.go

```diff
@@ -0,0 +1,361 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"go/types"
+	"math/rand"
+
+	"go.uber.org/mock/mockgen/model"
+	"golang.org/x/tools/go/packages"
+)
+
+type importModeParser struct {
+	pkgName string
+}
+
+func (p *importModeParser) parsePackage(packageName string, ifaces []string) (*model.Package, error) {
+	p.pkgName = packageName
+
+	pkg, err := p.loadPackage(packageName)
+	if err != nil {
+		return nil, fmt.Errorf("load package: %w", err)
+	}
+
+	interfaces, err := p.extractInterfacesFromPackage(pkg, ifaces)
+	if err != nil {
+		return nil, fmt.Errorf("extract interfaces from package: %w", err)
+	}
+
+	return &model.Package{
+		Name:       pkg.Types.Name(),
+		PkgPath:    packageName,
+		Interfaces: interfaces,
+	}, nil
+}
+
+func (p *importModeParser) loadPackage(packageName string) (*packages.Package, error) {
+	cfg := &packages.Config{
+		Mode: packages.NeedDeps | packages.NeedImports | packages.NeedTypes | packages.NeedTypesInfo | packages.NeedEmbedFiles,
+	}
+	pkgs, err := packages.Load(cfg, packageName)
+	if err != nil {
+		return nil, fmt.Errorf("load packages: %w", err)
+	}
+
+	if len(pkgs) != 1 {
+		return nil, fmt.Errorf("packages length must be 1: %d", len(pkgs))
+	}
+
+	if len(pkgs[0].Errors) > 0 {
+		errs := make([]error, len(pkgs[0].Errors))
+		for i, err := range pkgs[0].Errors {
+			errs[i] = err
+		}
+
+		return nil, errors.Join(errs...)
+	}
+
+	return pkgs[0], nil
```

Comment:
`pkgs[0].Errors` has type `[]packages.Error` instead of `[]error`. 
`errors.Join` received `error...`. `[]packages.Error` cannot be converted to `[]error`, even though `packages.Error` implements `error`.
This code - is conversion between these types.